app/frontend/src/lib/components/Header.svelte

```diff
@@ -0,0 +1,115 @@
+<script lang="ts">
+  import { Refresh } from "$wailsjs/go/main/App"
+  import { deviceLogs, getNextTheme, theme, toggleTheme, type CarbonTheme } from "$lib/state.svelte"
+
+  import { Header, HeaderGlobalAction, HeaderUtilities } from "carbon-components-svelte"
+
+  import { RefreshCcw, ToggleLeft } from "@lucide/svelte"
+
+  function formatTimestamp(time: Date) {
+    return new Date(time).toLocaleTimeString()
+  }
+
+  const themeLabels: Record<CarbonTheme, string> = {
+    g100: "G100",
+    white: "White",
+  }
+
+  let lastLog = $derived($deviceLogs?.length ? $deviceLogs[$deviceLogs.length - 1] : null)
+  let lastUpdatedTimestamp = $derived(lastLog ? formatTimestamp(lastLog.Time) : formatTimestamp(new Date()))
+  let isRefreshing = $state(false)
+  let refreshResetTimer: ReturnType<typeof setTimeout> | null = null
+
+  let nextTheme = $derived(getNextTheme($theme))
+  let currentThemeLabel = $derived(themeLabels[$theme])
+  let nextThemeLabel = $derived(themeLabels[nextTheme])
+  let themeTone = $derived($theme === "g100" ? "dark" : "light")
+
+  async function handleRefresh() {
+    if (isRefreshing) {
+      return
+    }
+
+    isRefreshing = true
+    if (refreshResetTimer) {
+      clearTimeout(refreshResetTimer)
+      refreshResetTimer = null
+    }
+
+    Refresh()
+
+    refreshResetTimer = setTimeout(() => {
+      isRefreshing = false
+      refreshResetTimer = null
+    }, 450)
+  }
+</script>
+
+<Header id="header" class="header" uiShellAriaLabel="USB tree status">
+  <span class="label">Last updated:</span>
+  <span class="timestamp">{lastUpdatedTimestamp}</span>
+  <HeaderUtilities class="utilities">
+    <HeaderGlobalAction
+      class="theme-action"
+      data-theme-tone={themeTone}
+      iconDescription="Dark mode"
+      kind="primary"
+      icon={ToggleLeft}
+      aria-label={`Switch to ${nextThemeLabel} theme (current ${currentThemeLabel})`}
+      onclick={toggleTheme}
+    />
+    <HeaderGlobalAction
+      class={`refresh-action${isRefreshing ? " refresh-action--spinning" : ""}`}
+      aria-label="Refresh"
+      icon={RefreshCcw}
+      kind="primary"
+      iconDescription="Refresh"
+      tooltipAlignment="end"
+      onclick={handleRefresh}
+    />
+  </HeaderUtilities>
+</Header>
+
+<style lang="scss">
+  @use "variables.scss" as *;
+
+  .label {
+    color: var(--color-header-text);
+    padding-right: $spacing-02;
+    font-weight: 600;
+  }
+
+  .timestamp {
+    color: var(--color-header-text);
+    font-weight: 400;
+  }
+
+  :global .theme-action :is(.bx--btn__icon, .lucide-icon) circle,
+  :global .theme-action :is(.bx--btn__icon, .lucide-icon) rect {
```

Comment:
https://svelte.dev/docs/svelte/global-styles#:global()

```suggestion
  :global(.theme-action :is(.bx--btn__icon, .lucide-icon) circle),
  :global(.theme-action :is(.bx--btn__icon, .lucide-icon) rect) {
```